### ISSUE TITLE
Getting a FileEntry by path

### ISSUE THREAD

**Issue by LoganDark** [NeekSandhu/react-aspen#12]
In my file watcher I'm performing some extra operations on certain files (when they're in a special folder, I do some parsing on them and set metadata so that the rendered items can display this data). I can't seem to get a hold of the FileEntry corresponding to some path.

I need to do this to update metadata when renames happen. My application uses atomic writes so I need to re-do the detection and parsing logic when something is renamed.

`model.root.inotify` seems to know what to do with paths so how come I can't do the same?

**Comment by LoganDark** [comment]
Maybe this is an XY problem and the real solution would be to allow me to modify metadata using inotify. Doesn't really matter in the end

**Comment by NeekSandhu** [comment]
Is`forceLoadFileentryAtPath` not returning you the handle?

Also since you say you are trying to get a handle after the rename, could it be that you are querying the old file by mistake?

Also note that if inotify is used properly, the FileEntry object reference remains the same even after it is renamed, again this is assuming inotify was provided with appropriate info.

**Comment by LoganDark** [comment]
> Is`forceLoadFileentryAtPath` not returning you the handle?

I... did not know that function existed.

> Also since you say you are trying to get a handle after the rename, could it be that you are querying the old file by mistake?

Considering I couldn't find out how to run any query, no.

> Also note that if inotify is used properly, the FileEntry object reference remains the same even after it is renamed, again this is assuming inotify was provided with appropriate info.

Exactly, this is why it's not being populated with the correct metadata.

I'll try out `forceLoadFileentryAtPath`, but I'd prefer not to force a load if aspen is not yet interested (i.e. parents haven't been expanded and such).

**Comment by LoganDark** [comment]
Turns out none of that was needed.

The solution I settled on was to check if, during a rename, the new path is a different "kind" of file than the old path (as in the metadata needs changing). If so, just delete the old path and use an Added event to create the new one.

The code looks something like this:

![image](https://user-images.githubusercontent.com/4723091/135790700-f0c445c3-2d17-4adc-ab2a-06c2ce7ce3f1.png)
